quic/state/test/AckHandlersTest.cpp

```diff
@@ -1107,6 +1113,74 @@ TEST_P(AckHandlersTest, ImplictAckEventCreation) {
       ackTime);
 }
 
+TEST_P(AckHandlersTest, TestRTTPacketObserverCallback) {
```

Comment:
We need to make sure that this works correctly in the case of out-of-order packet arrivals. What happens if I send packets 1 - 3, and then get packets 2, 3, 1? Will I get the ACKs as they arrive? Can you add a test with this?